### PR TITLE
NIAD-155 outbound sequence number manager record send

### PR DESCRIPTION
**Pull request by Alicja-Kainos**
Any suggestion how the saving would be done better according to best practices when it comes to python and dynamodb are welcome :)

### CODE REVIEW

File: nhais/outbound/state/work_description.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        """
+        if persistence_store is None:
+            raise ValueError('Expected persistence store')
+
+        self.persistence_store = persistence_store
+        self._deserialize_data(store_data)
+
+    async def publish(self):
+        """
+        Attempts to publish the local state of the work description to the state store, checks versions to avoid
```
Comment (grzegorz-robak) [inline nhais/outbound/state/work_description.py]: At what point it checks version?

File: nhais/outbound/state/work_description.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+def create_new_work_description(persistence_store: pa.PersistenceAdaptor,
+                                segments
+                                ) -> WorkDescription:
+    """
+    Builds a new local work description instance given the details of the message, these details are held locally
+    until a `publish` is executed
+    """
+
+    if persistence_store is None:
+        raise ValueError('Expected persistence store to not be None')
```
Comment (grzegorz-robak) [inline nhais/outbound/state/work_description.py]: The same check is done in `WorkDescription` constructor

**Review by grzegorz-robak — DISMISSED** [review]

File: nhais/outbound/state/work_description.py (hunk 2 above)
Comment (Alicja-Kainos) [reply nhais/outbound/state/work_description.py]: right, it should be removed from one place :)

File: nhais/outbound/state/work_description.py (hunk 1 above)
Comment (Alicja-Kainos) [reply nhais/outbound/state/work_description.py]: That is the case: I did not want to use the version checking like in MHS (get, than update). Anyway, the scope of the task is to save the state, not doing update, if I got it right :)

File: nhais/outbound/state/work_description.py (hunk 2 above)
Comment (Alicja-Kainos) [reply nhais/outbound/state/work_description.py]: done

File: nhais/outbound/state/work_description.py (hunk 1 above)
Comment (Alicja-Kainos) [reply nhais/outbound/state/work_description.py]: @mattd-kainos, @bartek-sarul , what do you think about it, any ideas? should we store the version in the table or do the versioning other way?

Comment (grzegorz-robak) [reply nhais/outbound/state/work_description.py]: Sure, my point wasn't to make versioning, but if it doesn't do that then remove versioning from description :)

Comment (Alicja-Kainos) [reply nhais/outbound/state/work_description.py]: removed XD

**Review by grzegorz-robak — APPROVED** [review]

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
Will need to be merged to integration-adaptor-nhais instead

**Comment by Alicja-Kainos** [comment]
changes moved to separate nhais repo;
nhais module from integration-adaptor repo is removed int this pull request (https://github.com/nhsconnect/integration-adaptors/pull/244#pullrequestreview-406372033)

**Pull request closed without merging**